### PR TITLE
[REVIEW] Fix chunksize issue with `DataFrame.to_csv`

### LINKED ISSUE #5588: [BUG] df.to_csv fails for certain chunksize values
**Describe the bug**
`df.to_csv` supports the `chunksize` argument to specify number of rows to write at a time. This fails for certain chunksize values w.r.t the df length.

**Steps/Code to reproduce bug**

```python
In [1]: import cudf                                                                                                                                                                                                                                                                                                                                                                                                                   
In [2]: import numpy as np         
In [3]: df = cudf.DataFrame({'a':np.arange(10_000)})                                                                                                                                                                                                                                                                                                                                                                                  
In [5]: df.to_csv("file.csv",chunksize=1000)                                                                                                                                                                                                                                                                                                                                                                                          
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-5-de508b13a746> in <module>
----> 1 df.to_csv("file.csv",chunksize=1000)

miniconda3/envs/rapids-15-jun26/lib/python3.7/site-packages/cudf/core/dataframe.py in to_csv(self, path, sep, na_rep, columns, header, index, line_terminator, chunksize)
   6109             index,
   6110             line_terminator,
-> 6111             chunksize,
   6112         )
   6113 

miniconda3/envs/rapids-15-jun26/lib/python3.7/contextlib.py in inner(*args, **kwds)
     72         def inner(*args, **kwds):
     73             with self._recreate_cm():
---> 74                 return func(*args, **kwds)
     75         return inner
     76 

miniconda3/envs/rapids-15-jun26/lib/python3.7/site-packages/cudf/io/csv.py in to_csv(df, path, sep, na_rep, columns, header, index, line_terminator, chunksize)
    132         header=header,
    133         line_terminator=line_terminator,
--> 134         rows_per_chunk=rows_per_chunk,
    135     )

cudf/_lib/csv.pyx in cudf._lib.csv.write_csv()

cudf/_lib/csv.pyx in cudf._lib.csv.write_csv()

RuntimeError: cuDF failure at: /opt/conda/envs/rapids/conda-bld/libcudf_1593132811419/work/cpp/src/io/csv/writer_impl.cu:403: Unexpected empty strings column.
```

```python
In [6]: df.to_csv("file.csv",chunksize=1500) # Works
```
**Expected behavior**
The writing succeeds for all `chunksize` values (provided there is enough GPU memory).

**Environment overview (please complete the following information)**
 - Method of cuDF install: Conda
   - Nightly from Jun26

### PR DESCRIPTION
**Pull request by rgsl888prabhu**
Due to split, it introduced an empty table with `0` rows which caused this issue. 

closes #5588

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by trevorsm7** [comment]
Not specifically related to this PR, but I noticed these nearby lines while reviewing: https://github.com/rapidsai/cudf/blob/2495992311595bfbe203e6395a9a00d1fe989acb/cpp/src/io/csv/writer_impl.cu#L517-L518
It took me a while to figure what was going on in the second line. I think `auto str_table_view = str_table_ptr->view();` would be much clearer. I'd suggest squeezing that change in along with this PR.

**Comment by rgsl888prabhu** [comment]
> Fix looks good.
> I wonder if [this line](https://github.com/rapidsai/cudf/blob/2495992311595bfbe203e6395a9a00d1fe989acb/cpp/src/io/csv/writer_impl.cu#L479) should be `auto n_chunks = util::div_rounding_up_safe(num_rows, n_rows_per_chunk) - 1;` though. Is this the reason why `split` is introducing an empty table at the end?

Kind of yes, it will create split with even where it end with actual number of  rows and split will always add end row whether it has been provided or not, so there will be an empty table.

File: python/cudf/cudf/tests/test_csv.py [hunk 1]
```diff
@@ -1542,3 +1542,16 @@ def test_csv_writer_empty_dataframe(tmpdir):
 
     assert df.shape == (0, 2)
     assert all(df.dtypes == ["object", "object"])
+
+
+def test_csv_write_chunksize_corner_case(tmpdir):
+    # With this num of rows and chunksize
+    # libcudf splits table such a way that it
+    # will end up creating an empty table slice
+    # which caused the issue.
```
Comment (davidwendt) [inline python/cudf/cudf/tests/test_csv.py:1551]: Can you add the issue number to this comment?

**Pull request merged**